sparse_strips/vello_common/src/glyph.rs

```diff
@@ -209,66 +205,50 @@ fn prepare_glyph_run<'a>(
     run: &GlyphRun<'a>,
     outlines: &OutlineGlyphCollection<'_>,
 ) -> PreparedGlyphRun<'a> {
-    // TODO: Consider extracting the scale from the glyph transform and applying it to the font size.
-    if !run.hint || run.glyph_transform.is_some() {
+    if !run.hint {
         return PreparedGlyphRun {
-            transform: run.transform,
-            glyph_transform: run.glyph_transform,
+            transform: run.transform * run.glyph_transform.unwrap_or(Affine::IDENTITY),
             size: Size::new(run.font_size),
-            scale: 1.0,
-            horizontal_skew: run.horizontal_skew,
             normalized_coords: run.normalized_coords,
             hinting_instance: None,
         };
     }
 
-    // Hinting doesn't make sense if we later scale the glyphs via some transform. So, if
-    // this glyph can be scaled uniformly, we extract the scale from its global and glyph
-    // transform and apply it to font size for hinting. Note that this extracted scale
-    // should be later applied to the glyph's position.
+    // We perform vertical-only hinting.
+    //
+    // Hinting doesn't make sense if we later scale the glyphs via some transform. So we extract
+    // the scale from the global transform and glyph transform and apply it to the font size for
+    // hinting. We do require the scaling to be uniform: simply using the vertical scale as font
+    // size and then transforming by the relative horizontal scale can cause, e.g., overlapping
+    // glyphs. Note that this extracted scale should be later applied to the glyph's position.
     //
-    // If the glyph is rotated or skewed, hinting is not applicable.
+    // As the hinting is vertical-only, we can handle horizontal skew, but not vertical skew or
+    // rotations.
 
-    // Attempt to extract uniform scale from the run's transform.
-    if let Some((scale, transform)) = take_uniform_scale(run.transform) {
-        let font_size = run.font_size * scale as f32;
+    let total_transform = run.transform * run.glyph_transform.unwrap_or(Affine::IDENTITY);
+    let [t_a, t_b, t_c, t_d, t_e, t_f] = total_transform.as_coeffs();
 
-        let size = Size::new(font_size);
+    let uniform_scale = t_a == t_d;
+    let vertically_uniform = t_b == 0.;
+
+    if uniform_scale && vertically_uniform {
+        let vertical_font_size = run.font_size * t_d as f32;
+        let size = Size::new(vertical_font_size);
         let hinting_instance =
             HintingInstance::new(outlines, size, run.normalized_coords, HINTING_OPTIONS).ok();
-
-        return PreparedGlyphRun {
-            transform,
-            glyph_transform: run.glyph_transform,
+        PreparedGlyphRun {
+            transform: Affine::new([1., t_b, t_c, 1., t_e, t_f]),
```

Comment:
```suggestion
            transform: Affine::new([1., 0., t_c, 1., t_e, t_f]),
```

nit: Setting the second parameter to `0.` signals to the reader that, as per prior lines, we know what this value will end up being; the variables we do specify here are the ones that are variable.